.github/workflows/publish.yml

```diff
@@ -17,32 +14,22 @@ jobs:
         with:
           fetch-depth: 1
 
-      - name: Read Python version
-        id: python-version
-        run: echo "version=$(cat .python-version)" >> $GITHUB_OUTPUT
-
-      - name: Read Poetry version
-        id: poetry-version
-        run: echo "version=$(cat .poetry-version)" >> $GITHUB_OUTPUT
-
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: ${{ steps.python-version.outputs.version }}
+          python-version-file: pyproject.toml
 
-      - name: Set up Poetry
-        run: |
-          python3 -m venv $POETRY_HOME
-          $POETRY_HOME/bin/pip install "poetry==${{ steps.poetry-version.outputs.version }}"
+      - name: Set up uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          version: 0.6.11
```

Comment:
Can't that replace `actions/setup-python` too? (One less step, faster runs...)

Reply:
It can! Ref: https://docs.astral.sh/uv/guides/integration/github/

But I opted not to, because of this comment in the docs:

> Alternatively, the official GitHub setup-python action can be used. This can be faster, because GitHub caches the Python versions alongside the runner.